### PR TITLE
[Tuna] Add fin to MIOpen CI

### PR DESCRIPTION
**Pull request by JehandadKhan**
Often changes to MIOpen break either Tuna or fin which makes it difficult to manage the versioning and coordinating changes. This PR makes fin part of MIOpen as a subtree to ensure breaking changes are caught the PR level.

cc @atamazov

### CODE REVIEW

**Comment by JehandadKhan** [comment]
@atamazov CI Passes

**Comment by atamazov** [comment]
@JehandadKhan Just in case: Have you considered the DeepCode warnings -- https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1044#issuecomment-885252582?

File: Dockerfile [hunk 1]
```diff
@@ -103,6 +103,7 @@ RUN cget -p $PREFIX install kitware/cmake@v3.15.1
 
 ADD min-requirements.txt /min-requirements.txt
 RUN CXXFLAGS='-isystem $PREFIX/include' cget -p $PREFIX install -f /min-requirements.txt
+RUN cget -p $PREFIX install nlohmann/json@350ff4f7ced7c4117eae2fb93df02823c8021fcb
```
Comment (atamazov) [inline Dockerfile:106]: [Q] Can we install this only if necessary? Like we do for MLIR tests?

File: Jenkinsfile [hunk 2]
```diff
@@ -314,6 +314,16 @@ pipeline {
                         buildHipClangJobAndReboot(setup_cmd: "", build_cmd: "", execute_cmd: execute_cmd, no_reboot:true)
                     }
                 }
+              stage('Fin Test') {
```
Comment (atamazov) [inline Jenkinsfile:317]: Maybe
```suggestion
              stage('Tuna Fin Build Test') {
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -67,6 +67,7 @@ set(MIOPEN_LOG_FUNC_TIME_ENABLE Off CACHE BOOL "")
 set(MIOPEN_ENABLE_SQLITE_BACKOFF On CACHE BOOL "")
 
 option( BUILD_DEV "Build for development only" OFF)
+option(MIOPEN_ENABLE_FIN "Enable the fin driver for MIOpen"  OFF)
```
Comment (atamazov) [inline CMakeLists.txt:70]: Maybe this way?
```suggestion
option( MIOPEN_ENABLE_FIN "Enable building the fin driver for Tuna" OFF)
```

**Review by atamazov — COMMENTED** [review]
@JehandadKhan This can be merged as is. Please ping me if you wish me to proceed with this.

**Pull request merged**